galsim/phase_psf.py

```diff
@@ -0,0 +1,1442 @@
+    obscuration disk (or to the pupil center if `obscuration = 0.`).  You can specify how many
+    struts there are (evenly spaced in angle), how thick they are as a fraction of the pupil
+    diameter, and what angle they start at relative to the positive y direction.
+
+    The size (in meters) and sampling interval (in meters) of the pupil plane array representing the
+    aperture can be set directly using the the `pupil_plane_size` and `pupil_plane_scale` keywords.
+    However, in most situations, it's probably more convenient to let GalSim set these automatically
+    based on the pupil geometry and the nature of the (potentially time-varying) phase aberrations
+    from which a PSF is being derived.
+
+    The pupil plane array physical size is by default set to twice the pupil diameter producing a
+    Nyquist sampled PSF image.  While this would always be sufficient if using sinc interpolation
+    over the PSF image for subsequent operations, GalSim by default uses the much faster (though
+    approximate) quintic interpolant, which means that in some cases -- in particular, for
+    significantly aberrated optical PSFs without atmospheric aberrations -- it may be useful to
+    further increase the size of the pupil plane array, thereby increasing the sampling rate of the
+    resulting PSF image.  This can be done by increasing the `oversampling` keyword.
+
+    The pupil plane array physical sampling interval (which is directly related to the resulting PSF
+    image physical size) is set by default to the same interval as would be used to avoid
+    significant aliasing (image folding) for an obscured Airy profile with matching diameter and
+    obscuration and for the value of `folding_threshold` in the optionally specified gsparams
+    argument.  If the phase aberrations are significant, however, the PSF image size computed this
+    way may still not be sufficiently large to avoid aliasing.  To further increase the pupil plane
+    sampling rate (and hence the PSF image size), you can increase the value of the `pad_factor`
+    keyword.  An additional way to set the pupil sampling interval for a particular set of phase
+    screens (i.e., for a particular PhaseScreenList) is to provide the screens in the `screen_list`
+    argument.  Each screen in the list computes its own preferred sampling rate and the
+    PhaseScreenList appropriately aggregates these. This last option also requires that a wavelength
+    `lam` be specified, and is particularly helpful for creating PSFs derived from turbulent
+    atmospheric screens.
+
+    Finally, when specifying the pupil geometrically, Aperture may choose to make a small adjustment
+    to `pupil_plane_scale` in order to produce an array with a good size for FFTs.  If your
+    application depends on knowing the size and scale used with the Fourier optics framework, you
+    can obtain these from the `Aperture.pupil_plane_size` and `Aperture.pupil_plane_scale`
```

Comment:
Recommend lowercase `aperture` here.  The capital A implies that these are class attributes, rather than instance attributes.